### PR TITLE
Viclim/rails5

### PR DESCRIPTION
**Pull request by viclim**
### Notable Change:
- Airbrake now requires `project_id` during initialization. Include `project_id` in `config/airbrake.yml`.
- Rails requires `secret_key_base` to be set.
- Use `puma` as the development server
- Add instruction to run elasticsearch through docker in README

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -7,26 +7,21 @@ end
 
 gem 'rails', '5.1.4'
 gem 'rails-controller-testing', '~> 1.0'
-
-gem 'rack-contrib', '~> 2.0.1'
```
Comment (nickmarden) [inline Gemfile]: Did you intend for these Gemfile changes to be part of this commit?

File: README.md [hunk 2]
```diff
@@ -10,7 +10,7 @@ The documentation on request parameters and response format is on the [API devel
 
 ### Ruby
 
-This code is currently tested against [Ruby 2.3.x](http://www.ruby-lang.org/en/downloads/).
+This code is currently tested against [Ruby 2.3](http://www.ruby-lang.org/en/downloads/).
```
Comment (nickmarden) [inline README.md]: Did you intend to revert this documentation change?

File: config/initializers/airbrake.rb [hunk 3]
```diff
@@ -50,7 +50,7 @@
   # Alternatively, you can integrate with Rails' filter_parameters.
   # Read more: https://goo.gl/gqQ1xS
   # c.blacklist_keys = Rails.application.config.filter_parameters
-end if Rails.env.production?
```
Comment (nickmarden) [inline config/initializers/airbrake.rb]: Why is this change necessary? I think that `if Rails.env.production?` and `unless Rails.env.development? || Rails.env.test?` are the same thing.

Comment (viclim) [reply config/initializers/airbrake.rb]: No `Rails.env.staging?`

Comment (nickmarden) [reply config/initializers/airbrake.rb]: No, there isn't a staging environment.

Comment (viclim) [reply config/initializers/airbrake.rb]: Reverted to using `if Rails.env.production?`.

File: README.md (hunk 2 above)
Comment (nickmarden) [reply README.md]: Ah, wait, I see the history now. It was `2.1` -> `2.3.x` -> `2.3`. This is fine.

File: Gemfile (hunk 1 above)
Comment (viclim) [reply Gemfile]: The gems were sorted alphabetically by rubocop. I reverted them to the original position.
The upgrades are intentional. I read through most of the changelogs before I upgrade them.
